stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -1,109 +1,147 @@
+import datetime as dt
+from _typeshed import Incomplete, ReadableBuffer, SupportsRead, SupportsReadline, SupportsTrunc, SupportsWrite, Unused
 from collections.abc import Callable, Iterable, Mapping, Sequence
 from types import TracebackType
-from typing import Any, TypeVar, overload
-from typing_extensions import Literal, Self, TypeAlias
+from typing import Any, NoReturn, Protocol, SupportsInt, TypeVar, overload, type_check_only
+from typing_extensions import Literal, Self, SupportsIndex, TypeAlias
 
-import psycopg2
-import psycopg2.extensions
 from psycopg2.sql import Composable
 
 _Vars: TypeAlias = Sequence[Any] | Mapping[str, Any] | None
+_AcceptedByInt: TypeAlias = str | ReadableBuffer | SupportsInt | SupportsIndex | SupportsTrunc
```

Comment:
I saw this alias used in many places in typeshed. Would it make sense to define it as a private attribute in `_typeshed`?

Reply:
Sure, a PR would be welcome, although the psycopg2 can only sensibly use it after most type checkers release a version that includes the new alias.